lua/orgmode/org/autocompletion/omni.lua

```diff
@@ -9,23 +9,38 @@ local data = {
   metadata = { 'DEADLINE:', 'SCHEDULED:', 'CLOSED:' },
 }
 
-local directives = { rgx = vim.regex([[^\#+\?\w*$]]), line_rgx = vim.regex([[^\#\?+\?\w*$]]), list = data.directives }
+local directives = {
+  line_rgx = vim.regex([[^\#\?+\?\w*$]]),
+  rgx = vim.regex([[^\#+\?\w*$]]),
+  list = data.directives,
+}
+
 local begin_blocks = {
-  rgx = vim.regex([[\(^\s*\)\@<=\#+\?\w*$]]),
   line_rgx = vim.regex([[^\s*\#\?+\?\w*$]]),
+  rgx = vim.regex([[\(^\s*\)\@<=\#+\?\w*$]]),
   list = data.begin_blocks,
 }
+
 local properties = {
   line_rgx = vim.regex([[\(^\s\+\|^\s*:\?$\)]]),
   rgx = vim.regex([[\(^\|^\s\+\)\@<=:\w*$]]),
+  extra_cond = function(line, _)
+    return not string.find(line, 'file:$')
+  end,
   list = data.properties,
 }
+
 local links = {
-  line_rgx = vim.regex([[\(\(^\|\s\+\)\[\[\)\@<=\(\*\|\#\)\?\(\w\+\)\?]]),
-  rgx = vim.regex([[\(\*\|\#\)\?\(\w\+\)\?$]]),
+  line_rgx = vim.regex([[\(\(^\|\s\+\)\[\[\)\@<=\(\*\|\#\|file:\)\?\(\w\+\)\?]]),
```

Comment:
You should update regexes to this:
```lua
  line_rgx = vim.regex([[\(\(^\|\s\+\)\[\[\)\@<=\(\*\|\#\|file:\)\?\(\(\w\|\/\|\.\|\\\)\+\)\?]]),
  rgx = vim.regex([[\(\*\|\#\|file:\)\?\(\(\w\|\/\|\.\|\\\)\+\)\?$]]),
```
It's a bit crazy, but it will allow autocompleting part of the filename.
For example, if you type this:

```
[[file:/home/user/orgfiles/t]]
```

It will autocomplete only fiels that start with `t`. Needs more changes below though, I leave the comment.

Reply:
I like it, this is a lot more intuitive.